fs/filter/filter_test.go

```diff
@@ -42,6 +42,28 @@ func testFile(t *testing.T, contents string) string {
 	return s
 }
 
+func TestNewFilterForbiddenMixOfFilesFromAndFilterRule(t *testing.T) {
+	Opt := DefaultOpt
+
+	// Set up the input
+	Opt.FilterRule = []string{"- filter1", "- filter1b"}
+	Opt.FilesFrom = []string{testFile(t, "#comment\nfiles1\nfiles2\n")}
+
+	rm := func(p string) {
+		err := os.Remove(p)
+		if err != nil {
+			t.Logf("error removing %q: %v", p, err)
+		}
+	}
+	// Reset the input
+	defer func() {
+		rm(Opt.FilesFrom[0])
+	}()
+
+	_, err := NewFilter(&Opt)
+	require.Error(t, err)
```

Comment:
You could add after this to make sure we get the right error

    require.Contains(t, err.Error(), "some substring from the error")

Reply:
Yes, it makes sense, I will add this assertion